oauth2_client/tests/test_client.py

```diff
@@ -1,84 +1,204 @@
 """
 Client tests.
+
+Local imports of model-related files are required for in-IDE test runs.
+Otherwise a model import is attempted before django.setup() call and
+an exception is thrown.
 """
-import time
+from datetime import timedelta
 
+import requests_mock
+import six
 from django.utils import timezone
+from oauthlib.oauth2 import TokenExpiredError
 
 from test_case import StandaloneAppTestCase
-from .test_compat import patch
+from .test_compat import Mock, patch
 
 
-class CreateAccessTokenTest(StandaloneAppTestCase):
+class ClientTest(StandaloneAppTestCase):
     """
-    Ouath2 client test cases.
-    Local imports of model-related files are required for in-IDE test runs.
-    Otherwise an import will be attempted before django.setup() call and
-    an exception is thrown.
+    OAuth2 client test cases.
     """
 
-    @patch('oauth2_client.client._get_token')
-    def test_no_token(self, get_token_mock):
+    @patch('oauth2_client.client.fetch_token')
+    def test_no_token(self, fetch_token_mock):
         """
         Ensure new token is created if there is no token.
         """
-        from oauth2_client.models import AccessToken
         from oauth2_client.client import get_client
-        from .factories import ApplicationFactory
+        from oauth2_client.models import AccessToken
+        from .factories import ApplicationFactory, fake_token
 
+        now = timezone.now()
         app = ApplicationFactory()
-        token = {'access_token': 'my_token', 'expires_at': time.time()}
-        get_token_mock.return_value = token
+        # we don't want this one stored in DB, so no Factory Boy here
+        token_from_provider = AccessToken(
+            application=app,
+            token=fake_token(),
+            expires=now + timedelta(seconds=AccessToken.TIMEOUT_SECONDS + 10),
+            created=now,
+        )
+        fetch_token_mock.return_value = token_from_provider
 
         client = get_client(app.name)
         self.assertIsNotNone(client)
-        access_token = AccessToken.objects.get(application=app)
-        self.assertEqual(token, access_token.token)
-        get_token_mock.assert_called_once_with(app)
+        token_from_db = AccessToken.objects.get(application=app)
+        self.assertEqual(token_from_provider.token, token_from_db.token)
+        self.assertEqual(
```

Comment:
Please fit into one line if it fits 120, it is much easier to read and review because that one line will have single semantic meaning.